### PR TITLE
Add doc for plan-preview feature

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Introducing plan-preview feature
```

### CODE REVIEW

File: docs/content/en/docs/user-guide/plan-preview.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+This feature will available for all application kinds: KUBERNETES, TERRAFORM, CLOUD_RUN, LAMBDA and Amazon ECS.
+
+![](/images/plan-preview-comment.png)
+<p style="text-align: center;">
+PlanPreview with GitHub actions <a href="https://github.com/pipe-cd/actions-plan-preview">pipe-cd/actions-plan-preview</a>
+</p>
+
+## Prerequisites
+
+- Your Pipeds are running at least version `v0.11.0`.
```
Comment (nakabonne) [inline docs/content/en/docs/user-guide/plan-preview.md:29]: nit but it's on you
```suggestion
- Your Pipeds are greater than at least `v0.11.0`.
```

**Comment by nakabonne** [comment]
left a nit comment but mostly LGTM
/lgtm

File: docs/content/en/docs/user-guide/plan-preview.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+title: "Plan preview"
+linkTitle: "Plan preview"
+weight: 14
+description: >
+  Enables the ability to preview the deployment plan against a given commit before merging.
+---
+
+> NOTE: This feature is currently under Alpha status.
+
+In order to help developers review the pull request with a better experience and more confidence to approve it to trigger the actual deployments,
```
Comment (khanhtc1202) [inline docs/content/en/docs/user-guide/plan-preview.md:11]:
```suggestion
In order to help developers review the pull request with a better experience and more confidence to approve it and trigger the actual deployments,
```

File: docs/content/en/docs/user-guide/plan-preview.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+You can run it locally or integrate it to your CI system to run automatically when a new pull request is opened/updated. Use `--help` to see more options.
+
+``` console
+pipectl plan-preview --help
+```
+
+## GitHub Actions
+
+If you are using GitHub Actions, you can seamlessly integrate our prepared [actions-plan-preview](https://github.com/pipe-cd/actions-plan-preview) to your workflows. This automatically comments the plan-preview result on the pull request when it is opened or updated. You can also trigger to run plan-preview manually by leave a comment `/pipecd plan-preview` on the pull request.
```
Comment (khanhtc1202) [inline docs/content/en/docs/user-guide/plan-preview.md:54]: `/pipecd plan-preview`

🎉

File: docs/content/en/docs/user-guide/plan-preview.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+> NOTE: This feature is currently under Alpha status.
+
+In order to help developers review the pull request with a better experience and more confidence to approve it to trigger the actual deployments,
+PipeCD provides a way to preview the deployment plan of all updated applications by that pull request.
+
+Here are what will be included currently in the result of plan-preview process:
+
+- what application will be deployed once the pull request got merged
+- what deployment strategy (QUICK_SYNC or PIPELINE_SYNC) will be used
+- what resources will be added, deleted, or modified
```
Comment (khanhtc1202) [inline docs/content/en/docs/user-guide/plan-preview.md:18]: nit, but up to you 🙆‍♀️ 
```suggestion
- which application will be deployed once the pull request got merged
- which deployment strategy (QUICK_SYNC or PIPELINE_SYNC) will be used
- which resources will be added, deleted, or modified
```

Comment (nghialv) [reply docs/content/en/docs/user-guide/plan-preview.md:18]: 👍

File: docs/content/en/docs/user-guide/plan-preview.md (hunk 1 above)
Comment (nghialv) [reply docs/content/en/docs/user-guide/plan-preview.md:29]: Let me change this to `Ensure the version of your Piped is at least v0.11.0`.

**Comment by nghialv** [comment]
Thank you. Applied them.

**Comment by nakabonne** [comment]
Nice. Can't wait to see on our website
/lgtm

**Comment by khanhtc1202** [comment]
Go go go 🎉 
/approve

**Pull request merged**